packages/headless/src/api/knowledge/answer-slice.ts

```diff
@@ -0,0 +1,45 @@
+import {
+  BaseQueryFn,
+  createApi,
+  FetchArgs,
+  fetchBaseQuery,
+  FetchBaseQueryError,
+  retry,
+} from '@reduxjs/toolkit/query';
+import {ConfigurationSection} from '../../state/state-sections';
+
+type StateNeededByAnswerSlice = ConfigurationSection;
+
+/**
```

Comment:
Here, we are not in the realm of "public function that implementers need to know about", so JSDoc not necessary (only used by UI-KIT dev)

Personally find these kind of comments problematic. 

We typically should not need to explain "here's how this function X fit with this other one Y to create end result  Z". They do not evolve well over time, and tend to get out of sync pretty quickly with how the software actually works when you fast forward some months/years.

You can keep it if you prefer. 

But I think given that a contributor must know how RTK query works at a base level (you have to read about it a bit if you are going to be touching these files anyway) what this function does is perfectly understandable without the comment.